MUI/src/main/java/mui/ManticoreRunner.java

```diff
@@ -42,13 +57,50 @@ public void callProc(String[] command) {
 				protected Object doInBackground() throws Exception {
 					ProcessBuilder pb = new ProcessBuilder(command);
 					try {
+						Msg.info(this, "called doInBackground");
 						Process p = pb.start();
 						BufferedReader reader =
 							new BufferedReader(new InputStreamReader(p.getInputStream()));
 						String line = "";
+						long prevtime = Instant.now().getEpochSecond();
+
 						while ((line = reader.readLine()) != null && !isTerminated) {
 							logArea.append(line);
 							logArea.append(System.lineSeparator());
+							if (Instant.now().getEpochSecond() - 2 > prevtime) { // >1s between updates
+								Msg.info(this, "attempting fetchstate");
+								prevtime = Instant.now().getEpochSecond();
+								try {
+									Socket stateSock = new Socket(host, port + 1); // port + 1 to get state server
+									InputStream stateInputStream = stateSock.getInputStream();
+									try {
+										byte[] curBytes = stateInputStream.readAllBytes();
+										StateOuterClass.StateList sl =
+											StateOuterClass.StateList.parseFrom(curBytes);
+										List<StateOuterClass.State> states =
+											sl.getStatesList();
+										if (states.size() > 0) {
+											ManticoreStateListModel newModel =
+												new ManticoreStateListModel();
+											for (StateOuterClass.State s : states) {
+												newModel.stateList.get(s.getType()).add(s);
+											}
+											stateListModel = newModel;
+											Msg.info(this,
+												Integer.toString(stateListModel.stateList.size()));
+											updateStateList();
+										}
+
+									}
+									catch (Exception e) {
+										Msg.info(this, e.toString());
+									}
+									stateSock.close();
+								}
+								catch (IOException se) {
+									Msg.info(this, se.toString());
+								}
+							}
```

Comment:
This is looking a bit complicated. Can you add some more descriptive contents with what is happening here and why this is needed? I see that we are "attempting fetchstate" and this is all happening in the background thread to update the state list, so this is where we read the protobuf message right? Also, can we not keep the socket connected when we've initialized the run and then just read from it during this loop and close it at the end?

Reply:
I'll clean up the intermediary steps (they're remnants of debugging) but we can't keep the socket connected - as per the PR that implemented the state server: https://github.com/trailofbits/manticore/pull/1620

>Very synchronous - Uses a socket server that creates a new connection every time the TUI polls for an update. Ideally, we'd open a pipe and just stream log/state messages through that. That said, I highly doubt a dozen-ish TCP packets per second are going to have any substantial performance impact, but we should profile this to make sure.